tests/runtime/pointers

```diff
@@ -83,3 +83,23 @@ NAME Pointer to pointer arithmetic and dereference
 PROG uprobe:./testprogs/array_access:test_ptr_array { $p = (int32 **)arg0; @x = **($p + 1); exit(); }
 EXPECT @x: 2
 AFTER ./testprogs/array_access
+
+NAME Pointer is truthy for if
+PROG BEGIN { if (curtask) { @ = 1; } exit(); }
+EXPECT @: 1
+
+NAME Pointer is truthy for tenary
+PROG BEGIN { @ = curtask ? 1 : 2; exit(); }
+EXPECT @: 1
+
+NAME Pointer is truthy for logical expressions
+PROG BEGIN { @ = (curtask && 1) ? 1 : 2; exit(); }
+EXPECT @: 1
+
+NAME Pointer is truthy for logical expressions
```

Comment:
```suggestion
NAME Pointer is falsey for logical expressions
```

Reply:
Or maybe the name should be "And-ing a truthy pointer with 0 is false"

Reply:
I've simplified this a bit, and made each case explain what it is intended to test.